### ISSUE TITLE
jsonSchemaOutputFormat drops constraints for array-form types (nullable objects)

### ISSUE THREAD

**Issue by KrishnaRMaddikara** [anthropics/anthropic-sdk-typescript#1212]
## Bug

`jsonSchemaOutputFormat()` keeps array-form `type` values but skips their type-specific transformations. For a nullable object, `properties`, `required`, and `additionalProperties` are moved into description text instead of being retained as schema constraints.

Reproduced against main at `1926adb4d292090975e6b5d19ebafe2274d2469e` (SDK 0.128.0):

```ts
import { jsonSchemaOutputFormat } from '@anthropic-ai/sdk/helpers/json-schema';

const schema = {
  type: 'object',
  properties: {
    result: {
      type: ['object', 'null'],
      properties: { name: { type: 'string' } },
      required: ['name'],
      additionalProperties: false,
    },
  },
  required: ['result'],
  additionalProperties: false,
} as const;

console.log(jsonSchemaOutputFormat(schema).schema);
```

Expected: `result` retains its properties, required field, and additional-properties restriction.
Actual: `result` keeps only its type union plus a description containing those rules. Nullable arrays and strings similarly lose supported item/minimum-length and format constraints.

Proposed fix: #1211. Its eight regression tests fail before the fix and pass afterward. No live API request has been made; this report concerns the SDK transformation. Related prior report in the Python SDK discussion: anthropics/anthropic-sdk-python#1764.
